pkg/driver/controllerserver.go

```diff
@@ -153,7 +167,10 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 
 	var deleteErr error
-	if prefix == "" {
+	if usePrefix, err := strconv.ParseBool(client.GetFSMetaField(meta, mounter.UsePrefix)); err == nil && usePrefix {
+		// UsePrefix is true, we do not delete anything
+		glog.V(4).Infof("Nothing to remove for %s", bucketName)
+	} else if prefix == "" {
```

Comment:
Nit: instead of the `else if` you could just `return &csi.DeleteVolumeResponse{}, nil` if there is nothing to remove.